### PR TITLE
#873 Add write_attributes_raw method

### PR DESCRIPTION
**Pull request by mdeweerd**
Implement #873 

Enables writing attributes that are not predefined.
Similar to read_attributes_raw

### CODE REVIEW

File: zigpy/zcl/__init__.py [hunk 1]
```diff
@@ -374,7 +374,14 @@ def _write_attr_records(
     async def write_attributes(
         self, attributes: Dict[Union[str, int], Any], manufacturer: Optional[int] = None
     ) -> List:
+        """Write attributes to device with internal 'attributes' validation"""
         args = self._write_attr_records(attributes)
+        return self.write_attributes_raw(args, manufacturer)
+
+    async def write_attributes_raw(
+        self, args: List[foundation.Attribute], manufacturer: Optional[int] = None
```
Comment (puddly) [inline zigpy/zcl/__init__.py:382]: `args` was an internal variable name but making it a parameter necessitates a more useful name.  Could you rename it to `attrs`?

Comment (mdeweerd) [reply zigpy/zcl/__init__.py]: I renamed it, but I have to review the test cases I added.

File: tests/test_zcl.py [hunk 2]
```diff
@@ -388,12 +388,28 @@ async def test_write_wrong_attribute(cluster):
         assert cluster._write_attributes.call_count == 1
 
 
+async def test_write_unknown_attribute(cluster):
+    with patch.object(cluster, "_write_attributes", new=AsyncMock()):
+        # Using an invalid attribute name, the call should fail
+        res = await cluster.write_attributes({"dummy_attribute": 5})
+        assert isinstance(res.exception(), KeyError)
```
Comment (puddly) [inline tests/test_zcl.py:395]:
```
with pytest.raises(KeyError):
    await cluster.write_attributes({"dummy_attribute": 5})
```

Comment (mdeweerd) [reply tests/test_zcl.py]: Thanks, this worked (tried another method as well that didn't).

**Comment by mdeweerd** [comment]
I've updated the branch.  The tests pass in my fork, but some executions fail because of uploads to https://coveralls.io/ .
There is also a minor change to the README.md because I wanted to see if that would trigger the workflow in my fork.

**Pull request merged**